admin_app/src/components/LanguageButtonBar.tsx

```diff
@@ -1,94 +1,131 @@
 import { Layout } from "@/components/Layout";
+import { useAuth } from "@/utils/auth";
 import {
-  DEFAULT_LANGUAGE,
-  LANGUAGE_OPTIONS,
   appColors,
   appStyles,
   sizes,
 } from "@/utils";
+import { apiCalls } from "@/utils/api";
 import { AddCircle } from "@mui/icons-material";
 import { Menu, MenuItem, ToggleButton, ToggleButtonGroup } from "@mui/material";
 import React from "react";
 
 interface Language {
-  code: string;
-  label: string;
+  language_id: number;
+  language_name: string;
 }
-
-const LanguageButtonBar = ({ expandable }: { expandable: boolean }) => {
-  const [langList, setLangList] = React.useState<(Language | undefined)[]>(
-    expandable
-      ? [LANGUAGE_OPTIONS.find((l) => l.code === DEFAULT_LANGUAGE)].filter(
-          Boolean,
-        )
-      : LANGUAGE_OPTIONS,
-  );
-  const [selectedLang, setSelectedLang] = React.useState<string>("en");
+interface LanguageButtonBarProps {
+  expandable: boolean;
+  onLanguageSelect: (language_id: number) => void;
+}
+const LanguageButtonBar = ({ expandable, onLanguageSelect }: LanguageButtonBarProps) => {
+  const [langList, setLangList] = React.useState<Language[]>([]);
+  const [selectedLang, setSelectedLang] = React.useState<number | undefined>(undefined);
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const { token } = useAuth();
+  React.useEffect(() => {
+    const fetchLanguages = async () => {
+      try {
+        const languages = await apiCalls.getLanguageList(token ?? '');
+        setLangList(languages);
+      } catch (error) {
+        console.error('Failed to fetch languages:', error);
+      }
+    };
+
+    const fetchDefaultLanguage = async () => {
+      try {
+        const defaultLanguage = await apiCalls.getDefaultLanguage(token ?? '');
+        setSelectedLang(defaultLanguage.language_id);
+        onLanguageSelect(defaultLanguage.language_id);
+      } catch (error) {
+        console.error('Failed to fetch default language:', error);
+      }
+    };
+
+    if (token) {
+      fetchLanguages();
+      fetchDefaultLanguage();
+    }
+  }, [token, expandable]);
```

Comment:
Why do we add `expandable` to this? It is not a state variable right and never changes?